### PR TITLE
Use Anthropic messages API

### PR DESCRIPTION
**Pull request by janimo**
This switches to the newer Anthropic API. The claude-3 models only work with this (so do claude-2.1 and claude-instant-1.2)

https://docs.anthropic.com/claude/reference/migrating-from-text-completions-to-messages

### CODE REVIEW

**Comment by merrymercy** [comment]
@janimo Great! Can you also add some if/else to make it also work with old versions or enforce a new version here? https://github.com/sgl-project/sglang/blob/08df63a6f8b9d27ae80cf705ab9496632c8a18c2/python/pyproject.toml#L25

**Comment by janimo** [comment]
@merrymercy thanks for the review. Updated the package version required. The only model that does not work with the new API is the legacy Claude 2.

File: examples/quick_start/anthropic_example_complete.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def few_shot_qa(s, question):
 \n\nAssistant: Rome
 """)
     s += "\n\nHuman: " + question + "\n"
-    s += "\n\nAssistant:" + sgl.gen("answer", stop="\n", temperature=0)
+    s += "\n\nAssistant:" + sgl.gen("answer", stop="\\n", temperature=0)
```
Comment (merrymercy) [inline examples/quick_start/anthropic_example_complete.py:22]: is "\\\n" a typo?

**Review by merrymercy — CHANGES_REQUESTED** [review]

File: examples/quick_start/anthropic_example_complete.py (hunk 1 above)
Comment (janimo) [reply examples/quick_start/anthropic_example_complete.py:22]: No, the new API apparently does not take empty stop sequences, and considers '\n' empty....

**Pull request merged**